src/admonitions/admonition.py

```diff
@@ -38,20 +89,31 @@ def in_codeblock(admonition: re.Match) -> bool:
                 for codeblock in codeblock_matches
             )
 
+        body_prefix: re.Pattern = None
+
         def convert_admonition(match: re.Match) -> str:
             if in_codeblock(match):
                 # don't convert
                 return match.group()
             alert_type: str = match.group("type")
+            alert_title: str = match.group("title").strip()
             alert_body: str = match.group("body")
 
             admonition_type = ADMONITION_TYPE_MAP.get(
                 alert_type.lower(), alert_type.lower()
             )
-            admonition_title = alert_type.title()
-            admonition_body = ALERT_BODY_PREFIX.sub("    ", alert_body.strip())
+            admonition_title = alert_title or alert_type.title()
+            admonition_body = body_prefix.sub("    ", alert_body.strip())
             return f'!!! {admonition_type} "{admonition_title}"\n{admonition_body}\n'
 
-        markdown = ALERT_PATTERN.sub(convert_admonition, markdown)
+        # convert basic syntax
+        codeblock_matches = list(GH_CODEBLOCK_PATTERN.finditer(markdown))
+        body_prefix = ALERT_BASIC_BODY_PREFIX
+        markdown = ALERT_BASIC_PATTERN.sub(convert_admonition, markdown)
+
+        # convert multiline blockquote syntax (GitLab only)
+        codeblock_matches = list(GH_CODEBLOCK_PATTERN.finditer(markdown))
```

Comment:
The conversions for GitHub style syntax shouldn't add/remove code-blocks. Am I missing something?

Reply:
No it won't, but it will change the codeblocks position (start and end indexes).
That's why I have to re-find them ;)

Reply:
I could have tried to write a very smart RegEx that supports both syntaxes in order to implement a single-pass _alert to admonition_ conversion, but the RegEx would have been ugly :sweat_smile: